libs/vscode/tasks/src/lib/nx-task-commands.ts

```diff
@@ -266,14 +261,28 @@ async function promptForRunMany() {
     return;
   }
 
-  const flags = await selectFlags('run-many', RUN_MANY_OPTIONS, 'nx');
+  let options = RUN_MANY_OPTIONS;
+  const projects = validProjectsForTarget(target);
+  if (projects && projects.length) {
+    options = [
+      {
+        name: 'projects',
+        type: OptionType.Array,
+        description: 'Projects to run',
+        aliases: [],
+        enum: projects,
+      },
+      ...RUN_MANY_OPTIONS,
+    ];
+  }
+
+  const flags = await selectFlags('run-many', options, 'nx', { target });
```

Comment:
Adding target to the userSetFlags parameter shows 'Execute: nx run-many --target=<target>` on the Command Palette instead of just `nx: run-many` with the selected options. Seems to make more sense as that's how you'd use the run-many command.